File: Design_Patterns/Observer_Design_pattern/cpp/MobileDisplay.cpp
/**
 * MobileDisplay.cpp
 * 
 * Implementation of MobileDisplay class
 */

#include "MobileDisplay.h"
#include <iostream>

MobileDisplay::MobileDisplay(WeatherStation* ws) : weatherStation(ws) {}

void MobileDisplay::update() {
    int currentTemp = weatherStation->getTemperature();
    display(currentTemp);
}

void MobileDisplay::display(int temperature) {
    std::cout << "📱 Mobile Display: Current temperature is " 
              << temperature << "°C" << std::endl;
}

File: Design_Patterns/BehaviouralDesignPattern/Observer_Design_pattern/java/Observer/MobileDisplay.java
package Observer;

import Observable.WeatherStation;

/**
 * MobileDisplay - Concrete Observer Implementation
 * 
 * This class represents a mobile device that displays weather information.
 * It observes the WeatherStation and updates its display when notified.
 * 
 * Key Features:
 * - Implements Observer interface
 * - Maintains reference to the WeatherStation (Observable)
 * - Updates display when notified of temperature changes
 */
public class MobileDisplay implements Observer {
    private WeatherStation weatherStation;

    public MobileDisplay(WeatherStation weatherStation) {
        this.weatherStation = weatherStation;
    }

    @Override
    public void update() {
        int currentTemp = weatherStation.getTemperature();
        display(currentTemp);
    }

    private void display(int temperature) {
        System.out.println("📱 Mobile Display: Current temperature is " +
                temperature + "°C");
    }
}
